src/autumnaton.ts

```diff
@@ -0,0 +1,217 @@
+    "leftleg1",
+    "rightarm1",
+    "rightleg1",
+    "cowcatcher",
+    "periscope",
+    "radardish",
+  ] as const;
+  // Libram doesn't have amount of leg upgrades specifically like the other types do
+  const legUpgrades = AutumnAton.currentUpgrades().filter((u) => u.includes("leg")).length;
+  const currentUpgrades = AutumnAton.currentUpgrades();
+  const acquirableUpgrades = profitRelevantUpgrades.filter(
+    (upgrade) => !currentUpgrades.includes(upgrade)
+  );
+  const currentBestLocation = maxBy(fullLocations, (loc: Location) => averageAutumnatonValue(loc));
+  const currentExpectedProfit =
+    averageAutumnatonValue(currentBestLocation) * expectedRemainingExpeditions();
+
+  const upgradeValuations = acquirableUpgrades.map((upgrade) => {
+    const upgradeLocations = fullLocations.filter(
+      (location) => getAutumnatonUniques(location)[0] === upgrade
+    );
+    const bestLocContainingUpg = maxBy(upgradeLocations, (loc: Location) =>
+      averageAutumnatonValue(loc)
+    );
+
+    if (upgrade === ("periscope" || "radardish")) {
+      const bestLocWithInstalledUpg = maxBy(fullLocations, (loc: Location) =>
+        averageAutumnatonValue(loc, AutumnAton.visualAcuity() + 1)
+      );
+      const extraExpectedProfit =
+        averageAutumnatonValue(bestLocContainingUpg) +
+        averageAutumnatonValue(bestLocWithInstalledUpg) *
+          Math.max(0, expectedRemainingExpeditions() - 1);
+
+      return { upgrade: upgrade, profit: extraExpectedProfit };
+    }
+    if (upgrade === ("rightleg1" || "leftleg1")) {
+      const bestLocWithInstalledUpg = currentBestLocation;
+      const extraExpectedProfit =
+        averageAutumnatonValue(bestLocContainingUpg) +
+        averageAutumnatonValue(bestLocWithInstalledUpg) *
+          Math.max(0, expectedRemainingExpeditions(legUpgrades + 1) - 1);
+
+      return { upgrade: upgrade, profit: extraExpectedProfit };
+    }
+    if (upgrade === ("rightarm1" || "leftarm1")) {
+      const bestLocWithInstalledUpg = maxBy(fullLocations, (loc: Location) =>
+        averageAutumnatonValue(loc, undefined, AutumnAton.zoneItems() + 1)
+      );
+      const extraExpectedProfit =
+        averageAutumnatonValue(bestLocContainingUpg) +
+        averageAutumnatonValue(bestLocWithInstalledUpg) *
+          Math.max(0, expectedRemainingExpeditions() - 1);
+
+      return { upgrade: upgrade, profit: extraExpectedProfit };
+    }
+    if (upgrade === "cowcatcher") {
+      const bestLocWithInstalledUpg = currentBestLocation;
+      const extraExpectedProfit =
+        averageAutumnatonValue(bestLocContainingUpg) +
+        (seasonalItemValue(bestLocWithInstalledUpg) +
+          averageAutumnatonValue(bestLocWithInstalledUpg)) *
+          Math.max(0, expectedRemainingExpeditions() - 1);
+
+      return { upgrade: upgrade, profit: extraExpectedProfit };
+    }
+    return { upgrade: upgrade, profit: 0 };
+  });
+  const mostValuableUpgrade = maxBy(upgradeValuations, ({ profit }) => profit);
```

Comment:
```suggestion
  const mostValuableUpgrade = maxBy(upgradeValuations, "profit");
```

Reply:
Ah I see, maxBy options,  `a key on the objects we're looking at`

Nice